src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneAnalyzerAdapterForQuerying.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+using System.IO;
+using System.Text;
+using Corax.Pipeline;
+using Lucene.Net.Analysis.Tokenattributes;
+using Analyzer = Corax.Analyzers.Analyzer;
+using LuceneAnalyzer = Lucene.Net.Analysis.Analyzer;
+
+namespace Raven.Server.Documents.Indexes.Persistence.Lucene;
+
+internal sealed unsafe class LuceneAnalyzerAdapterForQuerying : LuceneAnalyzerAdapter
+{
+    private LuceneAnalyzerAdapterForQuerying(LuceneAnalyzer analyzer,
+        delegate*<Analyzer, ReadOnlySpan<byte>, ref Span<byte>, ref Span<Token>, ref byte[], void> functionUtf8,
+        delegate*<Analyzer, ReadOnlySpan<char>, ref Span<char>, ref Span<Token>, void> functionUtf16) : base(analyzer, functionUtf8, functionUtf16)
+    {
+    }
+
+    private static void Run(Analyzer adapter, ReadOnlySpan<byte> source, ref Span<byte> output, ref Span<Token> tokens, ref byte[] buffer)
+    {
+        var @this = (LuceneAnalyzerAdapterForQuerying)adapter;
+        var analyzer = @this.Analyzer;
+
+        var data = Encoding.UTF8.GetString(source);
+        using (var reader = new StringReader(data))
+        {
+            int currentOutputIdx = 0;
+            var currentTokenIdx = 0;
+
+            using var stream = analyzer.TokenStream(null, reader);
```

Comment:
Note: Before we introduced the optimized adapter, we were using `ReusableTokenStream`, but I don't think we need to use it here because it will never be reused anyway. https://github.com/maciejaszyk/ravendb/blob/e0df629fa1943abcb7e9c09935025c1cf4eb1203/src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneAnalyzerAdapter.cs#L43

Reply:
Why will it never be reused? IIRC, that is stored as a thread static value, no?

Reply:
I looked into the Analyzer abstract, and it seems to be just a property.